src/main/java/org/folio/circulation/rules/cache/CirculationRulesCache.java

```diff
@@ -1,208 +1,123 @@
-  private final Map<String, Rules> rulesMap = new ConcurrentHashMap<>();
+  private Map<String, Rules> rulesMap = new ConcurrentHashMap<>();
 
   public static CirculationRulesCache getInstance() {
     return instance;
   }
 
   private CirculationRulesCache() {}
 
-  /**
-   * Completely drop the cache. This enforces rebuilding the drools rules
-   * even when the circulation rules haven't changed.
-   */
   public void dropCache() {
     rulesMap.clear();
   }
 
-  /**
-   * Enforce reload of the tenant's circulation rules.
-   * This doesn't rebuild the drools rules if the circulation rules haven't changed.
-   * @param tenantId  id of the tenant
-   */
-  public void clearCache(String tenantId) {
-    Rules rules = rulesMap.get(tenantId);
-    if (rules == null) {
-      return;
-    }
-    rules.reloadTimestamp = 0;
+  public void dropRulesForTenant(String tenantId) {
+    Rules rules = new Rules();
+    rulesMap.put(tenantId, rules);
   }
 
-  private boolean isCurrent(String tenantId, Rules rules) {
-    if (rules == null) {
-      log.info("Rules object is null considering it not current for tenant {}", tenantId);
-      return false;
-    }
-
-    long currentTimestamp = System.currentTimeMillis();
-    boolean isCurrent = rules.reloadTimestamp + MAX_AGE_IN_MILLISECONDS > currentTimestamp;
-    log.info("Rules object is current for tenant {}: {}. " +
-        "Reload timestamp is {}, current timestamp is {}",
-      tenantId, isCurrent, rules.reloadTimestamp, currentTimestamp);
-    return isCurrent;
-  }
-
-  /**
-   * Reload is needed if the last reload is TRIGGER_AGE_IN_MILLISECONDS old
-   * and a reload hasn't been initiated yet.
-   * @param rules - rules to reload
-   * @return whether reload is needed
-   */
-  private boolean reloadNeeded(String tenantId, Rules rules) {
-    if (rules.reloadInitiated) {
-      log.info("Rules reload is already initiated for tenant {}", tenantId);
-      return false;
-    }
-
-    long currentTimestamp = System.currentTimeMillis();
-    boolean reloadNeeded = rules.reloadTimestamp + TRIGGER_AGE_IN_MILLISECONDS < currentTimestamp;
-    log.info("Rules reload is needed for tenant {}: {}. " +
-        "Reload timestamp is {}, current timestamp is {}",
-      tenantId, reloadNeeded, rules.reloadTimestamp, currentTimestamp);
-    return reloadNeeded;
-  }
+  private boolean rulesExist(String tenantId) {
+    Rules rules = rulesMap.get(tenantId);
 
-  private boolean rebuildNeeded(String tenantId, Rules rules) {
-    long currentTimestamp = System.currentTimeMillis();
-    boolean rebuildNeeded = rules.rebuildTimestamp + DROOLS_OBJECT_LIFETIME_IN_MILLISECONDS <
-      currentTimestamp;
-    log.info("Drools object rebuild is needed for tenant {}: {}. " +
-        "Rebuild timestamp is {}, current timestamp is {}",
-      tenantId, rebuildNeeded, rules.rebuildTimestamp, currentTimestamp);
-    return rebuildNeeded;
+    return rules != null && !rules.getRulesAsText().isEmpty();
```

Comment:
Can `rules.getRulesAsText()` be null too?

Reply:
We now may not even need the second check condition, since we've eliminated the cache clear method.  The cache effectively now only has two states: either it's completely uninitialized, in which case the entire thing is a null value, or it has processed rules in it, in which case the second condition can never be true.